eZ/Bundle/EzPublishLegacyBundle/Cache/Switchable.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * This file is part of the eZ Publish Kernel package
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishLegacyBundle\Cache;
+
+trait Switchable
+{
+    /** @var bool */
+    private $enabled = true;
+
+    public function switchOn()
+    {
+        $this->enabled = true;
+    }
+
+    public function switchOff()
+    {
+        $this->enabled = false;
+    }
+
+    /**
+     * Returns the value of the switch
+     * @return bool
+     */
+    public function getSwitch()
```

Comment:
You're right. It actually had a better name before, but I've changed it to `$enabled` to make it "compatible" with the already existing `setEnabled`/`setDisabled` methods the `PersistenceCachePurger` had. But since it was changed to be a proper switchable, I can now name it properly.